File: FMI/ANUL II/SEM 2/MPP/ProiectJava/Zboruri/ZboruriModel/src/main/java/ro/mpp2024/domain/Angajat.java
package ro.mpp2024.domain;

public class Angajat extends Entity<String> {
    private String Username;
    private String Nume;
    private String Parola;

    public Angajat(String username, String parola) {
        this(username, "", parola);
    }

    public Angajat(String username, String nume, String parola) {
        this.Username = username;
        this.Nume = nume;
        this.Parola = parola;
    }

    public String getUsername() {
        return Username;
    }

    public void setUsername(String username) {
        this.Username = username;
    }

    public String getNume() {
        return Nume;
    }

    public void setNume(String nume) {
        this.Nume = nume;
    }

    public String getParola() {
        return Parola;
    }

    public void setParola(String parola) {
        this.Parola = parola;
    }

    @Override
    public String toString() {
        return "Angajat{" +
                "username='" + Username + '\'' +
                ", nume='" + Nume + '\'' +
                ", parola='" + Parola + '\'' +
                '}';
    }
}

File: FMI/ANUL II/SEM 2/MPP/ProiectC#/Zboruri/Model/Angajat.cs
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace Model
{
    [Serializable]
    public class Angajat: Entity<string>
    {
        public Angajat(string username, string nume, string parola)
        {
            Username = username;
            Nume = nume;
            Parola = parola;
        }
        
        [JsonConstructor]
        public Angajat(string username, string parola) : this(username, "", parola) {}
        
        [Key]
        public string Username { get; set; }
        public string Nume { get; set; }
        public string Parola { get; set; }

        public override string ToString()
        {
            return $"{nameof(Username)}: {Username}, {nameof(Nume)}: {Nume}";
        }
    }
}